Tcell instrumentation

zyme opened the pull request:
When we go to appengine, we'll lose our nginx WAF.  So we'll run [tcell RASP](https://earlyaccess.rapid7.com/tcell/) instead.  Still needs some work in terms of how the configs get rendered/bundled in circle.

As per @adrazhi, no need to redirect `tcell.log` off-container since what we rely on for tcell is API-based pushes to tcell.

We could update develop docs to run tcell locally.  I welcome opinions on that.

zyme committed:
added exclusions for tcell logs and config

zyme committed:
added rendering for tcell config

zyme committed:
Cleaned up some todos, added suggestions for how to incorporate tcell config

zyme committed:
Adding tcell config and jars

yufengwng commented on pepper-apis/src/appengine/app.yaml:
```diff
@@ -3,16 +3,14 @@ runtime: java11
-# get size from env var, try this without mx, ms
-entrypoint: java -Dddp.firecloudKeysDir=fc_keys -Dconfig.file=application.conf DataDonationPlatform.jar
+# will need to move all contents of tcell into deploy, and copy tcell_agent.config after rending
+entrypoint: java -javaagent:tcell/tcellagent.jar-Dddp.firecloudKeysDir=fc_keys -Dconfig.file=application.conf DataDonationPlatform.jar
```
Is it just me, or does checking in jars and dynamic libs into source seems a bit odd? Is there no way to manage this, say, through maven? Can we push this into docker image builds?

zyme replied:
It is unusual, but I haven't found a supported mvn repo.   I like the idea to push it into docker image for builds and keep it out of "source".

zyme replied:
However...I'm not sure if this plays nice with GAE.  Maybe we could put the jars into buckets, then grab those as part of deploy?   CC @MocanaAtBroad

zyme replied:
After a bit of fiddling, I'm going to try to put the tcell tarball into a bucket (sans secrets) and then fetch that in circle.  Should be reproducible as long as we version that tarball accordingly and don't update it.

yufengwng replied:
🎉 woohoo!

MocanaAtBroad replied:
What I have done in the past with libraries that are not in Maven is to install them into local repo and then refer to them in Maven build as any other library.
Trick was to setup a module under main pom that when it got built it would grab the library and executed a maven install on it.
The solution here appears to work and this is just a runtime dependency, so maybe we leave it alone?

zyme committed:
Updating path to match new home for pepper tcell keys

zyme committed:
enabling tcell branch, trying out fetching tcell tarball from gcp bucket

zyme committed:
enabling tcell branch, trying out fetching tcell tarball from gcp bucket

zyme committed:
enabling tcell branch, trying out fetching tcell tarball from gcp bucket

zyme committed:
enabling tcell branch, trying out fetching tcell tarball from gcp bucket

zyme committed:
enabling tcell branch, trying out fetching tcell tarball from gcp bucket

zyme committed:
enabling tcell branch, trying out fetching tcell tarball from gcp bucket

zyme committed:
enabling tcell branch, trying out fetching tcell tarball from gcp bucket

zyme committed:
enabling tcell branch, trying out fetching tcell tarball from gcp bucket

zyme committed:
enabling tcell branch, trying out fetching tcell tarball from gcp bucket

MocanaAtBroad commented on .circleci/config.yml:
```diff
@@ -17,6 +17,25 @@ executors:
+            fi
+            TAR_FILE_PATH="/tmp/$(basename -- $TAR_FILE_URL)"
+            gsutil -u broad-ddp-prod cp  $TAR_FILE_URL $TAR_FILE_PATH
+            cd ~/repo/ddp-workspace
```
Going to have to run this, but maybe this is meant for ddp-angular? I don't think we have a ddp-workspace directory in `ddp-study-server` repo.

MocanaAtBroad replied:
This path works?

zyme replied:
Oops, let me  clean this up.

raiks2 committed:
DDP-4511-add-db-migration-for-isreadonly: added a Liquibase file (#149)

Co-authored-by: Sergei Retsia <sretsia@broadinstitute.org>

zyme commented:
@MocanaAtBroad @yufengwng after some review of various other options (put in docker, host in bucket), I think we should keep this as-is.  It's simple and easy to manage, and source-code-ish.  Okay if I merge up?

zyme commented:
@SimoneMaiwald just a heads up on some tcell backend stuff in the context of appengine.

zyme committed:
Merge branch 'study-server-to-appengine' into tcell-instrumentation

# Conflicts:
#	.circleci/config.yml
#	pepper-apis/src/appengine/DataDonationPlatform.yaml

zyme committed:
Prep tcell artifacts

zyme committed:
Prep tcell artifacts

zyme committed:
Prep tcell artifacts

zyme committed:
Prep tcell artifacts

zyme committed:
Prep tcell artifacts

zyme committed:
Prep tcell artifacts

zyme committed:
Prep tcell artifacts

zyme committed:
Prep tcell artifacts

ssettipalli committed:
DDP-4512 . thank-you email fix for about-you

yufengwng commented on .circleci/config.yml:
```diff
@@ -21,9 +21,26 @@ executors:
+    working_directory: ~/repo/pepper-apis
 
 commands:
+  install-tcell:
+    steps:
+      - run:
+          name: Download tcell files from bucket
```
Is the plan to simply check tcell files into source? So we don't need this CircleCI command anymore?

zyme replied:
In the plan-of-the-moment, we'd just grab them from a bucket.  In which case we can delete the files here :)

zyme committed:
Prep tcell artifacts

yufengwng commented on .circleci/config.yml:
```diff
@@ -21,9 +21,26 @@ executors:
+      - run:
+          name: Download tcell files from bucket
+          command: |
+            vault read -format=json  secret/pepper/prod/v1/conf | jq -r .data.gcp.serviceKey | gcloud auth activate-service-account --key-file=-
```
We're reading the `prod` secrets in vault. Does that mean CircleCI have access to our prod secrets? That's a bit concerning...

_Edit: I think this was how things were setup originally. But now that I'm taking a second look at this, I'm a bit concerned, especially because most of our configs/secrets are in that same vault path..._

yufengwng replied:
Is there a way to restrict a vault token to reading certain paths? Maybe we should put the service account in a separate path so we isolate build/deploy secrets from actual application secrets? Also, I understand that this is a bit tangential to tcell, so I'm okay pushing this up and discussing this point separately.

yufengwng committed:
pex: first-class support for values and comparison (#130)

yufengwng committed:
pex: add profile birth date query (#145)

ssettipalli committed:
DDP-4514 . mailing address epoch fix

yufengwng committed:
DDP-4141 add dob validation for age-up (#123)

ssettipalli committed:
DDP-4516 . MBC followup fix

yufengwng committed:
DDP-4378 api: rename activity name and title (#80)

yufengwng committed:
Refactor user profile (#148)

ssettipalli committed:
MBC user enrollment and followup activity fix

raiks2 committed:
DDP 4134 auth0 client id to be scoped to tenant

yufengwng committed:
DDP-4458 reconcile schema for utf8mb4 (#155)

* DDP-4458 reconcile schema for utf8mb4

* update existing liquibase fails so they don't fail utf8mb4

yufengwng committed:
data exporter cli: add flag to do all elastic indices (#159)

ssettipalli committed:
DDP-4534 Add blood release & tissue release notifications (#164)

bskinner committed:
Updates the name used for the Auth0 Salt client (#162)

yufengwng committed:
angio: update placeholder for other_cancer_list diagnosis year (#167)

yufengwng committed:
Update pull request template to be more checklist-oriented (#97)

yufengwng committed:
Changes for consolidating database schemas (#156)

yufengwng committed:
Update API doc version (#171)

Co-authored-by: Blake Skinner <bskinner@broadinstitute.org>

dependabot[bot] committed:
Bump checkstyle from 8.18 to 8.29 in /ddp-automation (#75)

Bumps [checkstyle](https://github.com/checkstyle/checkstyle) from 8.18 to 8.29.
- [Release notes](https://github.com/checkstyle/checkstyle/releases)
- [Commits](https://github.com/checkstyle/checkstyle/compare/checkstyle-8.18...checkstyle-8.29)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

MocanaAtBroad committed:
Back-end build/test/deploy with parallel tests (#168)

* Build tests and deploy study server and housekeeping server using CircleCi.
* Support for running tests in parallel over multiple separate servers.
* Enable DDP build test/staging/prod promotion model based on Github branch assignments.
* Different specialized Dockerfiles defined to support building, testing, and deploying servers.
* Removal of Firecloud from code base.
* Modification of configuration file generation to allow for running tests from CircleCi.
* Modification of configuration file generation to allow deployment to GAE.

zyme committed:
Updated with GAE-compatible version of tcell

zyme committed:
Updated with GAE-compatible version of tcell

zyme committed:
Updated with GAE-compatible version of tcell

zyme committed:
tcell logs should go to stdout because in GAE you can't write to a local file

zyme committed:
Adding more metadata for tcell so it's easier to see what's what in the tcell dashboard

zyme committed:
Merge branch 'develop' into tcell-instrumentation

# Conflicts:
#	.circleci/config.yml
#	pepper-apis/Dockerfile-build
#	pepper-apis/api-build.sh
#	pepper-apis/config/manifest.rb
#	pepper-apis/config/testing-inmemorydb.conf.ctmpl
#	pepper-apis/parent-pom.xml
#	pepper-apis/pom.xml
#	pepper-apis/src/appengine/DataDonationPlatform.yaml
#	pepper-apis/src/appengine/Housekeeping.yaml
#	pepper-apis/src/main/java/org/broadinstitute/ddp/DataDonationPlatform.java
#	pepper-apis/src/test/java/org/broadinstitute/ddp/db/TransactionWrapperTest.java
#	pepper-apis/src/test/java/org/broadinstitute/ddp/util/TestDataSetupUtil.java

zyme committed:
merge typo

zyme committed:
Adding tcell to housekeeping

zyme committed:
Added tcell config--lost it in the merge.

MocanaAtBroad committed:
Update package to latest. Removes minimist vulnerability (#160)

MocanaAtBroad committed:
Ddp 4499 auth0 ip restrictions (#158)

- Implemented a new "whitelist" filter
- applied to the /register route,
- added support in templates to extract from vault the whitelisted ips
- new tests
- a little bit of refactoring of other filters used from DataDonationPlatform.

MocanaAtBroad committed:
Update script and configuration to handle changes currently in develop (#178)

ssettipalli committed:
DDP-4552 . MBC migration related fixes & improvements

ssettipalli committed:
DDP-4469 . Add activity instance guid to elastic structured index

zyme committed:
Added todo

zyme committed:
Merge branch 'develop' into tcell-instrumentation
